src/interfaces/coral_web/src/components/Agents/CreateAgentForm.tsx

```diff
@@ -0,0 +1,141 @@
+    model: MODEL_COMMMAND_R_PLUS,
+    tools: [],
+  });
+
+  const fieldErrors = {
+    ...(isAgentNameUnique(fields.name) ? {} : { name: 'Assistant name must be unique' }),
+  };
+
+  const canSubmit = (() => {
+    const { name, deployment, model } = fields;
+    const requredFields = { name, deployment, model };
+    return Object.values(requredFields).every(Boolean) && !Object.keys(fieldErrors).length;
+  })();
+
+  const handleChange = (key: Omit<AgentFormFieldKeys, 'tools'>, value: string) => {
+    setFields({
+      ...fields,
+      [key as string]: value,
+    });
+  };
+
+  const handleToolToggle = (toolName: string, checked: boolean) => {
+    const enabledTools = [...(fields.tools ? fields.tools : [])];
+    setFields({
+      ...fields,
+      tools: checked ? [...enabledTools, toolName] : enabledTools.filter((t) => t !== toolName),
+    });
+  };
+
+  const handleOpenSubmitModal = () => {
+    open({
+      title: `Create ${fields.name}?`,
+      content: (
+        <SubmitModalContent
+          agentName={fields.name}
+          onSubmit={handleSubmit}
+          isSubmitting={isSubmitting}
+          onClose={close}
+        />
+      ),
+    });
+  };
+  const handleSubmit = async () => {
+    if (!canSubmit) return;
+
+    try {
+      setIsSubmitting(true);
+      const agent = await createAgent(fields);
+      setFields({
+        name: '',
+        description: '',
+        preamble,
+        deployment: '',
+        model: '',
+        tools: [],
+      });
+      close();
+      setIsSubmitting(false);
+      router.push(`/agents?id=${agent.id}`, undefined, { shallow: true });
+    } catch (e) {
+      setIsSubmitting(false);
+      close();
+      error('Failed to create assistant');
+      console.error(e);
+    }
+  };
+
+  return (
+    <div className="relative h-full w-full">
+      <div className="flex h-full max-w-[650px] flex-col gap-y-2 overflow-scroll p-10 pb-32">
+        <Text styleAs="h4">Create an Assistant</Text>
+        <Text className="text-volcanic-700">
+          Create an unique assistant and share with your org
+        </Text>
+        <AgentForm
+          fields={fields}
+          onChange={handleChange}
+          onToolToggle={handleToolToggle}
+          errors={fieldErrors}
+          className="mt-6"
+        />
+      </div>
+      <div className="absolute bottom-0 right-0 flex w-full justify-end border-t border-marble-400 bg-white px-4 py-8">
+        <Button kind="green" splitIcon="add" onClick={handleOpenSubmitModal} disabled={!canSubmit}>
+          Create
+        </Button>
+      </div>
+    </div>
+  );
+};
+
+const SubmitModalContent: React.FC<{
+  agentName: string;
+  isSubmitting: boolean;
+  onSubmit: () => void;
+  onClose: () => void;
+}> = ({ agentName, isSubmitting, onSubmit, onClose }) => (
+  <div className="flex flex-col gap-y-20">
+    <Text>
+      Your {agentName} is about be visible publicly. Everyone in your organization will be able to
```

Comment:
nit: I know this is from designs, but I believe this text will not always read properly. For example if we name the assistant "Finances first quarter 2024" this will read oddly

Reply:
maybe we can say something like:

`Your assistant "{agentName}" is about...`